src/graph/transform/cuda/knn.cu

```diff
@@ -307,15 +307,18 @@ __global__ void BruteforceKnnShareKernel(
   FloatType* data_buff = SharedMemory<FloatType>();
   FloatType* query_buff = data_buff + block_size * feature_size;
   FloatType* dist_buff = query_buff + block_size * feature_size;
-  IdType* res_buff = reinterpret_cast<IdType*>(dist_buff + block_size * k);
+  IdType* res_buff = reinterpret_cast<IdType*>(
+      dist_buff + block_size * k);  // TODO: mem alignment?
   FloatType worst_dist = std::numeric_limits<FloatType>::max();
 
   // initialize dist buff with inf value
   for (auto i = 0; i < k; ++i) {
-    dist_buff[threadIdx.x * k + i] = std::numeric_limits<FloatType>::max();
+    dist_buff[threadIdx.x + i * block_size] =
+        std::numeric_limits<FloatType>::max();
   }
 
   // load query data to shared memory
+  // TODO: could be better here to exploit coalesce global memory access.
```

Comment:
could you help file an issue for this optimization so that we could track it? And add the issue ID here.

Reply:
See #5538

Reply:
pls add this ticket number into the comment: `TODO[5538]: ...`